packages/activemodel-adapter/lib/system/active-model-serializer.js

```diff
@@ -290,9 +290,10 @@ var ActiveModelSerializer = RESTSerializer.extend({
     }
   },
   typeForRoot: function(key) {
-    return camelize(singularize(key)).replace(/(^|\:)([A-Z])/g, function(match, separator, chr) {
+    var normalized = camelize(singularize(key)).replace(/(^|\:)([A-Z])/g, function(match, separator, chr) {
       return match.toLowerCase();
     }).replace('::', '/');
+    return this._super(normalized);
```

Comment:
If `_super` is not `RESTAdapter` the return would not be present. It is, seems ok, but is not safe if the parent class is not as expected.